third_party/gpus/find_rocm_config.py

```diff
@@ -69,36 +69,28 @@ def _get_header_version(path, name):
 
 def _find_rocm_config(rocm_install_path):
 
-  def rocm_version_numbers_pre_rocm50(path, prior_err):
-    version_file = os.path.join(path, ".info/version-dev")
-    if not os.path.exists(version_file):
-      raise ConfigError("{} ROCm version file ".format(prior_err) +
-                        '"{}" not found either.'.format(version_file))
-    version_numbers = []
-    with open(version_file) as f:
-      version_string = f.read().strip()
-      version_numbers = version_string.split(".")
-    major = int(version_numbers[0])
-    minor = int(version_numbers[1])
-    patch = int(version_numbers[2].split("-")[0])
-    return major, minor, patch
+  def rocm_version_numbers(path):
+    possible_version_files = [
+        "include/rocm-core/rocm_version.h",  # ROCm 5.2
+        "include/rocm_version.h",  # ROCm 5.1 and prior
```

Comment:
If I read the change correctly, actually this should be ROCm 5.0 and 5.1
Because it seems pre ROCm 5.0 there was a different version file, which is now not supported anymore?

Reply:
The file is the same, but some paths for headers files changed when going from ROCm 5.0 to 5.1 and then 5.1 to 5.2. If you plan on switching your internal CI, then it's best to move to ROCm5.2 or 5.3. Try to avoid ROCm5.1 with Tensorflow

Reply:
Thanks for confirming, I also thought it would be best to directly move to 5.3. And I guess I can do that after this PR has been merged.

Reply:
@akuegel  Finally, we have this merged! Thanks for you all your help here!

Reply:
I tried moving our build to ROCM 5.3, but then ran into this error:

```
@local_config_rocm//rocm:rocblas-hsaco: illegal output file name 'find: 'rocm/lib/rocblas/lib/library': No such file or directory' in rule @local_config_rocm//rocm:rocblas-hsaco: invalid target name 'find: 'rocm/lib/rocblas/lib/library': No such file or directory': target names may not contain ':'
```

This seems to indicate that it failed to find this directory when running the make_copy_dir rule in rocm_configure.bzl, but I am not quite sure. Maybe you have seen a similar error before and know what the problem is?
I also noticed that you deleted many make_copy_dir calls in your PR, but not this one. Why is this still needed, but not the others?

Reply:
Thank you for your help, deleting that make_copy_dir_rule seems to have fixed the issue :)

Reply:
@jayfurmanek @rahulbatra85

I noticed by now that by mistake I didn't roll forward the attempt to move our Kokoro to ROCM 5.3. I did this again today, and ran into a different error now:

```
this rule is missing dependency declarations for the following files included by 'tensorflow/compiler/xla/stream_executor/rocm/[rocm_helpers.cu.cc](https://www.google.com/url?q=http://rocm_helpers.cu.cc&sa=D)':
  '/opt/rocm/hip/include/hip/hip_version.h'
  '/opt/rocm/hip/include/hip/hip_runtime.h'
  '/opt/rocm/hip/include/hip/hip_common.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_hip_runtime.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_hip_common.h'
  '/opt/rocm/hip/include/hip/hip_runtime_api.h'
  '/opt/rocm/hip/include/hip/amd_detail/host_defines.h'
  '/opt/rocm/hip/include/hip/driver_types.h'
  '/opt/rocm/hip/include/hip/texture_types.h'
  '/opt/rocm/hip/include/hip/channel_descriptor.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_channel_descriptor.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_hip_vector_types.h'
  '/opt/rocm/hip/include/hip/surface_types.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_hip_runtime_pt_api.h'
  '/opt/rocm/hip/include/hip/amd_detail/hip_ldg.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_hip_atomic.h'
  '/opt/rocm/hip/include/hip/hip_vector_types.h'
  '/opt/rocm/hip/include/hip/amd_detail/device_library_decls.h'
  '/opt/rocm/hip/include/hip/amd_detail/llvm_intrinsics.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_device_functions.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_surface_functions.h'
  '/opt/rocm/hip/include/hip/amd_detail/ockl_image.h'
  '/opt/rocm/hip/include/hip/amd_detail/texture_fetch_functions.h'
  '/opt/rocm/hip/include/hip/hip_texture_types.h'
  '/opt/rocm/hip/include/hip/amd_detail/texture_indirect_functions.h'
  '/opt/rocm/hip/include/hip/amd_detail/amd_math_functions.h'
  '/opt/rocm/hip/include/hip/library_types.h'
```

Any idea what could be the reason?
It seems previously we have added hip/include to inc_dirs:

```
inc_dirs.append(rocm_config.rocm_toolkit_path + "/hip/include")
```

Now we add several other directories instead. I wonder whether we need to adjust this rule somehow?

```
make_copy_dir_rule(
            repository_ctx,
            name = "rocm-include",
            src_dir = rocm_toolkit_path + "/include",
            out_dir = "rocm/include",
            exceptions = ["gtest", "gmock"],
        ),
```

Reply:
> Any idea what could be the reason?

I guess there are symbolic links here. See also #35122 and #37006. If so, the resolution is to use `realpath`.

Reply:
I tried with realpath, but that did not help. But what seems to help is to still add inc_dirs.append(rocm_config.rocm_toolkit_path + "/hip/include") even for the new ROCM version (in addition to the other directories).
Not sure whether that would break the AMD community build though.

Reply:
FYI, now with https://github.com/tensorflow/tensorflow/commit/d21b5574c580ad8d58850d142fbf9984403e22c0 our Kokoro also uses ROCM 5.3
The title of that commit is wrong, forgot to update it when I rolled forward.